dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/helpers/calculate-initial-utilisation-and-fixed-fee.ts

```diff
@@ -0,0 +1,58 @@
+import { calculateInitialUtilisation, isDate } from '@ukef/dtfs2-common';
+import { TfmFacilitiesRepo } from '../../../../../repositories/tfm-facilities-repo';
+import { calculateFixedFee } from './calculate-fixed-fee';
+
+/**
+ * checks that all required values are present
+ * checks that cover start and end dates are in date format
+ * @param value
+ * @param interestPercentage
+ * @param dayCountBasis
+ * @param coverStartDate
+ * @param coverEndDate
+ * @returns true if all values are present and in the correct format
+ */
+export const hasRequiredValues = (
+  value: number,
+  interestPercentage: number,
+  dayCountBasis: number,
+  coverStartDate: string | Date | null,
+  coverEndDate: string | Date | null,
```

Comment:
Would this work? Ignore if not

```ts
coverStartDate?: Date,
coverEndDate?: Date,
```

Reply:
it does not work as ? is undefined and it is expecting null